### ISSUE TITLE
How to use source-XYZ ?

### ISSUE THREAD

**Issue by JohnsonWQ** [ghettovoice/vuelayers#139]
hey,mate, thx for vuelayers. i use XYZ-source to get a custom XYZ layer, but it didn't show up. i was confused about those properties, example: 'extent', 'resolutions' . specialily **tileGrid**, did i  use them in the wrong way ? or ....?  and could you plese give me some advice about it? 

version:   vue 2.5+, vuelayers 0.12
```vue
<template>
    <div class="map">
      <vl-map ref="map">
        <vl-view :zoom.sync="zoom" :center.sync="center" :resolutions.sync="resolutions" :extent="extent" :min-zoom="0" :max-zoom="9"></vl-view>
      
        <vl-layer-tile>
          <!-- <vl-source-osm></vl-source-osm> -->
          <vl-source-xyz :url="baseLayerUrl" :tileGrid="tileGrid" :crossOrigin="'anonymous'"></vl-source-xyz>
        </vl-layer-tile>
        
      </vl-map> 
    </div>
</template>

<script>
export default {
  data() {
    return {
      zoom: 3,
      center: [
        (54269.246678183685 + 72614.133466626) / 2,
        (37595.864092105185 + 46768.30748632633) / 2
      ],
      tileGrid: {
        extent: [
          41901.13637496065,
          26361.08674061112,
          94421.03308142163,
          69152.74024583958
        ],
        origin: [-272140, 276660],
        tileSize: [256, 256],
        resolutions: [
          132.2919312505292,
          52.91677250021167,
          26.458386250105836,
          13.229193125052918,
          5.291677250021167,
          2.6458386250105836,
          1.3229193125052918,
          0.5291677250021167,
          0.26458386250105836,
          0.13229193125052918
        ]
      },
      resolutions: [
        132.2919312505292,
        52.91677250021167,
        26.458386250105836,
        13.229193125052918,
        5.291677250021167,
        2.6458386250105836,
        1.3229193125052918,
        0.5291677250021167,
        0.26458386250105836,
        0.13229193125052918
      ],
      extent: [
        41901.13637496065,
        26361.08674061112,
        94421.03308142163,
        69152.74024583958
      ]
    };
  },
  name: "",
  components: {
    "top-nav": Top
  },
  computed: {
    baseLayerUrl() {
      return `http://192.168.34.198:6080/arcgis/rest/services/ZNJT/ODMAP/MapServer/tile/{z}/{y}/{x}`;
    }
  },
```

**Comment by ghettovoice** [comment]
Hi @JohnsonWQ ,
properties of vl-* components are pretty much the same like in OpenLayers library
because VueLayers is heavily based on it. So while something is missed in the VueLayers
documentation, you can find some tips in OpenLayers API documentation https://openlayers.org/en/latest/apidoc/module-ol_View-View.html. Especially if you already worked with it.

 As for your example, I suppose that you use som If so e non-default projection, not a EPSG:3857.
Am I right? If so, firstly you should register it in the registry and the provide its code to the `vl-view`
component. Here is an example how you can register [link](https://github.com/ghettovoice/vuelayers/issues/127#issuecomment-455993129) and [link](https://github.com/ghettovoice/vuelayers-demo/blob/vl-0.10.x/src/App.vue#L243) and [here](https://jsfiddle.net/ghettovoice/Ludfq6ay/).
In short: you should define projection, register it and set it to `vl-view` (to apply it for all map by default) or for a layer source (to apply it only for this source)

**Comment by JohnsonWQ** [comment]
Thanks for your reply. As u say, i'm not use the mainstream coordinate system, at least not EPSG3857.  But i can show it with native ol method when i define "extend,zoom,resolutions"  in the "view"  and  create "XYZ source" by "url,titleGrid".  It just show up, i'm not defined any custom coordinate system，just use ol default.  So that's what i am confused. I wondering if i am put those "properties" in the wrong place?  I just can't figure out what's the difference  with "vuelaysers" method or properties.

**Comment by ghettovoice** [comment]
It would be helpful if you add a jsfiddle or add code snippet here how you made this with OpenLayers. So I could translate it to the VueLayers.
